src/pairwise.jl

```diff
@@ -0,0 +1,313 @@
+function _pairwise!(::Val{:none}, f, dest::AbstractMatrix, x, y, symmetric::Bool)
+    @inbounds for (i, xi) in enumerate(x), (j, yj) in enumerate(y)
+        symmetric && i > j && continue
+
+        # For performance, diagonal is special-cased
+        if f === cor && eltype(dest) !== Union{} && i == j && xi === yj
+            # TODO: float() will not be needed after JuliaLang/Statistics.jl#61
+            dest[i, j] = float(cor(xi))
+        else
+            dest[i, j] = f(xi, yj)
+        end
+    end
+    if symmetric
+        m, n = size(dest)
+        @inbounds for j in 1:n, i in (j+1):m
+            dest[i, j] = dest[j, i]
+        end
+    end
+    return dest
+end
+
+function check_vectors(x, y, skipmissing::Symbol)
+    m = length(x)
+    n = length(y)
+    if !(all(xi -> xi isa AbstractVector, x) && all(yi -> yi isa AbstractVector, x))
```

Comment:
```suggestion
    if !(all(xi -> xi isa AbstractVector, x) && all(yi -> yi isa AbstractVector, y))
```
?

Reply:
Woops. I'll always be surprised by the number of tests one needs to write to ensure all possible mistakes are covered! I've added another one.